### ISSUE TITLE
Deprecation Warning: mglearn.datasets.make_forge()

### ISSUE THREAD

**Issue by yltang52** [amueller/introduction_to_ml_with_python#113]
The following warning message appeared when running `mglearn.datasets.make_forge()`:

    DeprecationWarning: Function make_blobs is deprecated; Please import make_blobs directly from scikit-learn
      warnings.warn(msg, category=DeprecationWarning)

Does it mean `mglearn` needs to update?

**Comment by amueller** [comment]
That's a bit strange. What's ``mglearn.__version__``?

**Comment by yltang52** [comment]
'0.1.7'

**Comment by yltang52** [comment]
OK, this seems to be the problem: the following commit is not available for `pip`.

    https://github.com/amueller/introduction_to_ml_with_python/commit/b54511da61b4283e415951d92feba70100914324#diff-9fe83ed3a2e00bc60f7eda44fd00c89e

**Comment by amueller** [comment]
Thanks for investigating, I'll have to update the pip version. There's really no problem though.

**Comment by roblemos** [comment]
I'm still getting this error as well. Actually, I think the deprecation error is being produced because you need to replace:
`from sklearn.datasets import make_blobs`
with
`from sklearn.datasets.samples_generator import make_blobs`
in `make_forge()`. YMMV.

**Comment by amueller** [comment]
@roblemos installing ``mglearn`` from pip?
I didn't get around to updating the pip version since February, this repo includes the fix.

**Comment by roblemos** [comment]
Yep, I installed from pip as part of a docker container. Good to know.

**Comment by Abuba-oss** [comment]
Actually I had the same problem when I` import mglearn` and I have tried using your : 
`from sklearn.datasets.samples_generator import` `make_blobs` but it won't work either and I wonder what i did wrong?

`import mglearn`
`import numpy as np `
`import matplotlib.pyplot as plt `
`import pandas as pd `
`from pandas.plotting import scatter_matrix`
`from sklearn.datasets import load_iris'
`from sklearn.datasets.samples_generator import make_blobs`
`from sklearn.model_selection import train_test_split`

`iris_dataset = load_iris()`


`X_train,X_test,y_train,y_test = train_test_split(`
    `iris_dataset['data'], iris_dataset['target'],random_state=0`
)


print('X_train shape: {}'.format(X_train.shape))
print('y_train shape : {}'.format(y_train.shape))

print('X_test shape :{}'.format(X_test.shape))
print('y_test shape :{}'.format(y_test.shape))


#Create dataframe from data in x_train
#label the columns using the string in iris_dataset.feature_names
iris_dataframe = pd.DataFrame(X_train,columns=iris_dataset.feature_names)
#Creatw a scatter matrix from the dataframe, color by y_train

`grr = scatter_matrix(iris_dataframe, c=y_train,figsize=(15,15), marker='o',`
                          ` hist_kwds={'bins':20}, s=60, alpha=.8,cmap=mglearn.cm3)`



:\Users\AIMLExpert\AppData\Local\Programs\Python\Python37\lib\site-packages\sklearn\externals\six.py:31: DeprecationWarning: The module is deprecated in version 0.21 and will be removed in version 0.23 since we've dropped support for Python 2.7. Please rely on the official version of six (https://pypi.org/project/six/).
  "(https://pypi.org/project/six/).", DeprecationWarning)
C:\Users\AIMLExpert\AppData\Local\Programs\Python\Python37\lib\site-packages\sklearn\externals\joblib\__init__.py:15: DeprecationWarning: sklearn.externals.joblib is deprecated in 0.21 and will be removed in 0.23. Please import this functionality directly from joblib, which can be installed with: pip install joblib. If this warning is raised when loading pickled models, you may need to re-serialize those models with scikit-learn 0.21+.
  warnings.warn(msg, category=DeprecationWarning)

**Comment by amueller** [comment]
@abuba-oss those are actually different deprecation warnings for the newer versions of scikit-learn.
We're doing a release this week, I'll try to update the book material after that.
Right now the book is lagging behind a bit. So there's nothing you're doing wrong and there's also no problem. You can just ignore the warnings for now.

**Comment by amueller** [comment]
Should be fixed now, please let me know if you run into any more issues.

**Comment by mozenmozen** [comment]
when trying:
`import mglearn`
`X,y=mglearn.datasets.make_forge()`

I got:
`/Users/admin/opt/anaconda3/lib/python3.7/site-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function make_blobs is deprecated; Please import make_blobs directly from scikit-learn
  warnings.warn(msg, category=FutureWarning)`

I assume this is fine and no need to worry about?

**Comment by amueller** [comment]
@mozenmozen yes that's fine. You can update mglearn to get rid of the warning though.

**Comment by farhanbadr09** [comment]
install **pip install scikit-learn**
install **pip install mglearn**
after installing those packages import **from sklearn.datasets import make_blobs**
then import **X,y=mglearn.datasets.make_forge()**
that's it

**Comment by amueller** [comment]
@farhanbadr09 can you elaborate? Are you encountering the issue still?

**Comment by sepideh68** [comment]
@amueller I am getting the same warning on Jupiter!
```
anaconda3/lib/python3.9/site-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function make_blobs is deprecated; Please import make_blobs directly from scikit-learn
  warnings.warn(msg, category=FutureWarning)
```

`mglearn.__version__` : '0.1.9'
`sklearn.__version__` : '1.0.2'

**Comment by amueller** [comment]
Should be fixed now in the update of mglearn.
